File: tests/parsers/samples/code/php/dynamic.php
<?php

namespace Samples;

function run_dynamic(string $name = 'summarize'): string
{
    if (!function_exists(__NAMESPACE__ . "\\$name")) {
        return 'unknown';
    }
    $callable = __NAMESPACE__ . "\\$name";
    return 'dynamic:' . $callable();
}

File: tests/parsers/samples/code/scala/dynamic.scala
package samples

object DynamicMetrics {
  private val registry: Map[String, () => String] = Map(
    "summarize" -> (() => SimpleMetrics.summarize())
  )

  def runDynamic(name: String = "summarize"): String =
    registry.get(name).map(fn => s"dynamic:${fn()}").getOrElse("unknown")
}
